Update Portworx Integration: metric names and readme (#2974)
DataDog/integrations-extras: portworx/README.md

                  "metrics": ["px_*"]
                }
              ]
            }
          }
  - componentNames:
    - Stork
    workloadConfigs:
    - workloadNames:
      - stork
      annotations:
        ad.datadoghq.com/stork.checks: |-
          {
            "openmetrics": {
              "instances": [
                {
                  "openmetrics_endpoint": "http://%%host%%:9091/metrics",
                  "namespace": "portworx",
                  "metrics": ["stork_*"]
                }
              ]
            }
          }
  - componentNames:
    - Autopilot
    workloadConfigs:
    - workloadNames:
      - autopilot
      annotations:
        ad.datadoghq.com/autopilot.checks: |-
          {
            "openmetrics": {
              "instances": [
                {
                  "openmetrics_endpoint": "http://%%host%%:9628/metrics",
                  "namespace": "portworx",
                  "metrics": ["autopilot_*"]
                }
              ]
            }
          }
```

At this point, the Datadog Agent is configured to scrape Portworx metrics and the Portworx, Stork, and Autopilot pods are annotated for Autodiscovery.

> You can verify the Agent is scraping metrics by running `kubectl exec <datadog-agent-pod> -n datadog-ns -- agent status` and looking for `portworx` in the `Checks` section.

Reviewer: This sentence duplicates the Validation section that follows immediately below (lines 152-160).

```suggestion
At this point, the Datadog Agent is configured to scrape Portworx metrics and the Portworx, Stork, and Autopilot pods are annotated for Autodiscovery.
```

To configure rules, see the [Autopilot documentation][12].

### Validation

Run the following command to confirm the Agent is scraping Portworx metrics:

```shell

> kubectl exec <datadog-agent-pod> -n <datadog-namespace> -- agent status

Reviewer: The Kubernetes namespace is referenced inconsistently across this section: literal `datadog-ns` on lines 24 and 148, but `<datadog-namespace>` placeholder here and on lines 192 and 196. Pick one convention (either treat the namespace as a placeholder everywhere, or use a literal example (and explain it once)) and stick with it.

Author: Hi @domalessi 
I believe this is consistent already.
Line no: 156, 191,195 use the placeholder.
Line no. 24 is a complete yaml example - with the explanation on Line no 33, to replace the namespace name as deemed fit.

- Monitor health and performance of your Portworx Cluster
- Track disk usage, latency and throughput for Portworx volumes

> - Setup Autopilto Rules https://docs.portworx.com/portworx-enterprise/operations/scale-portworx-cluster/autopilot

Reviewer: Three issues on this line: typo "Autopilto" → "Autopilot", "Setup" should be the verb form "Set up", and the URL needs to be a markdown link (bare URLs aren't standard in Datadog docs). The Autopilot URL is already defined as reference `[12]` later in the file, so you can reuse that.

```suggestion
- Set up [Autopilot rules][12]
```

## Setup

### Installation

#### Step 1 - Create Datadog credentials Secret

> Create a Kubernetes Secret with Metrics API and a Datadog API key

Reviewer: "Create a Kubernetes Secret with Metrics API and a Datadog API key" is unclear — "with Metrics API" doesn't parse. Also missing an ending period, and a colon would help introduce the YAML block.

```suggestion
Create a Kubernetes Secret containing your Datadog API and application keys:
```

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: datadog-credentials
  namespace: datadog-ns
  labels:
    app: autopilot
    component: datadog-provider
type: Opaque
data:
  # Base64-encoded Datadog API Key
  api-key: <base64-encoded-api-key>
  # Base64-encoded Datadog Application Key
  app-key: <base64-encoded-app-key>
```

- Replace the namespace, secret name, and encoded key values as required.

> - Ensure Autopilot has RBAC permission to read this Secret.

Reviewer: Per the Datadog style guide, replace "Ensure" with softer phrasing — "Ensure" implies a guarantee. Recasting as a direct imperative reads better here.

```suggestion
- Grant Autopilot RBAC permission to read this Secret.
```

> - The `app-key` must have metrics Read permission.

Reviewer: For parallel structure with the imperative bullets above ("Replace...", "Grant..."), recast this bullet as imperative as well. Also, "Read" should be lowercase unless it refers to a specific Datadog UI permission name.

```suggestion
- Use an application key with metrics read permission for `app-key`.
```

> #### Step 2 - Configure the Datadog Agent to export PX metrics

Reviewer: "PX" is Portworx jargon. Use the canonical product name "Portworx" consistently in docs (this affects multiple lines — flagging here and on subsequent occurrences).

```suggestion
#### Step 2 - Configure the Datadog Agent to export Portworx metrics
```

**2.1** Create a Datadog Agent values file (for example, `datadog_config.yaml`):

```yaml
datadog:
  site: "datadoghq.com"
  clusterName: "your-cluster-name"
  apiKeyExistingSecret: "datadog-credentials"
  kubelet:
    tlsVerify: false
  clusterChecks:
    enabled: true
  orchestratorExplorer:
    enabled: true
clusterAgent:
  enabled: true
  admissionController:
    enabled: true
    mutateUnlabelled: false
```

> Use your own `site`, `clusterName`, and `apiKeyExistingSecret`. The Agent scrapes PX metrics and sends them to Datadog.

Reviewer: "PX metrics" → "Portworx metrics" for canonical terminology.

```suggestion
Use your own `site`, `clusterName`, and `apiKeyExistingSecret`. The Agent scrapes Portworx metrics and sends them to Datadog.
```

Author: done

> **2.2** Install the Datadog Agent via Helm:

Reviewer: Per the Datadog style guide, replace "via" with "with" or "using".

```suggestion
**2.2** Install the Datadog Agent with Helm:
```

```shell
helm repo add datadog https://helm.datadoghq.com
helm upgrade --install datadog-agent datadog/datadog -f ./datadog_config.yaml
```

### Configuration

#### Step 2.3 - Annotate PX, Stork, and Autopilot pods

> Configure Datadog auto-discovery annotations so the Agent knows which endpoints to scrape. The typical Prometheus-style service endpoints are:

Reviewer: The Datadog feature is named "Autodiscovery" (one word, capitalized). Same fix applies on line 151.

```suggestion
Configure Datadog Autodiscovery annotations so the Agent knows which endpoints to scrape. The typical Prometheus-style service endpoints are:
```

Author: done

| Component | Endpoint | Metrics filter |
|-----------|----------|----------------|
| Portworx API | `http://%%host%%:17001/metrics` | `px_*` |
| Stork | `http://%%host%%:9091/metrics` | `stork_*` |
| Autopilot | `http://%%host%%:9628/metrics` | `autopilot_*` |

> **Note:** Portworx has many metrics. For most use cases, Portworx API metrics alone are sufficient.

> Apply a `ComponentK8sConfig` CR to add the annotations:

Reviewer: ```suggestion
Apply a `ComponentK8sConfig` Custom Resource (CR) to add the annotations:
```

Author: done

              "instances": [
                {
                  "openmetrics_endpoint": "http://%%host%%:17001/metrics",
                  "namespace": "portworx",
                  "metrics": ["px_*"]
                }
              ]
            }
          }
  - componentNames:
    - Stork
    workloadConfigs:
    - workloadNames:
      - stork
      annotations:
        ad.datadoghq.com/stork.checks: |-
          {
            "openmetrics": {
              "instances": [
                {
                  "openmetrics_endpoint": "http://%%host%%:9091/metrics",
                  "namespace": "portworx",
                  "metrics": ["stork_*"]
                }
              ]
            }
          }
  - componentNames:
    - Autopilot
    workloadConfigs:
    - workloadNames:
      - autopilot
      annotations:
        ad.datadoghq.com/autopilot.checks: |-
          {
            "openmetrics": {
              "instances": [
                {
                  "openmetrics_endpoint": "http://%%host%%:9628/metrics",
                  "namespace": "portworx",
                  "metrics": ["autopilot_*"]
                }
              ]
            }
          }
```

> At this point, the Datadog Agent is configured to scrape PX metrics and the PX, Stork, and Autopilot pods are annotated for auto-discovery.

Reviewer: Two terminology fixes on this line: "PX metrics" → "Portworx metrics" (twice) and "auto-discovery" → "Autodiscovery".

```suggestion
At this point, the Datadog Agent is configured to scrape Portworx metrics and the Portworx, Stork, and Autopilot pods are annotated for Autodiscovery.
```

Author: done

You can verify the Agent is scraping metrics by running `kubectl exec <datadog-agent-pod> -n datadog-ns -- agent status` and looking for `portworx` in the `Checks` section.

However, Autopilot is not yet configured to use the Datadog provider.

#### Step 3 - Configure the Autopilot Datadog provider in StorageCluster

Enable the Datadog provider in your `StorageCluster` spec under the `autopilot` section:

```yaml
spec:
  autopilot:
    enabled: true
    providers:
    - name: datadog
      type: datadog
      params:
        url: https://datadoghq.com
        secretName: datadog-credentials
        secretNamespace: datadog-ns
  monitoring:
    prometheus:
      enabled: false
      exportMetrics: true
```

Key points:
- `type: datadog` tells Autopilot to use the Datadog provider.
- `secretName`/`secretNamespace` must match the Secret created in Step 1.

> - Both Datadog and Prometheus are supported simultaneously as well as independently. However, an AutopilotRule configured for one provider cannot interchangeably work with the other.

Reviewer: "interchangeably work" is awkward. Recast for clarity.

```suggestion
- Both Datadog and Prometheus are supported simultaneously as well as independently. However, an AutopilotRule cannot be reused across providers — each rule is bound to one provider.
```

Author: removed

- If the Datadog Secret is updated, the Autopilot deployment must be restarted.

Apply the updated `StorageCluster` spec and wait for Autopilot to roll out with Datadog support.

To configure Autopilot rules, see the [Autopilot documentation][12].

### Validation

> Run the following command to confirm the Agent is scraping PX metrics:

Reviewer: "PX metrics" → "Portworx metrics" for canonical terminology.

```suggestion
Run the following command to confirm the Agent is scraping Portworx metrics:
```
